lib/restrictions.js

```diff
@@ -193,34 +196,33 @@ export async function get({id, explain = false} = {}) {
 * Gets restrictions given a zone ID and a resource ID.
  *
  * @param {object} options - Options to use.
- * @param {string} options.zone - The ID of the zone that the
- *   restriction applies to.
- * @param {string} options.resource - The ID of the resource that the
- *   restriction applies to.
+ * @param {object} options.query - The query.
  * @param {boolean} [options.explain=false] - An optional explain boolean.
  *
  * @returns {Promise<object | ExplainObject>} Resolves with an object with the
  *   `restrictions` property which is an array of matching records or an
  *   ExplainObject if `explain=true`.
  */
-export async function getAll({zone, resource, explain = false} = {}) {
-  const query = {
-    'restriction.zone': zone,
-    'restriction.resource': resource
-  };
+export async function getAll({query, explain = false} = {}) {
```

Comment:
Note: Updated to pass `query` instead of `zone` and `resource` just to make it consistent with the other getAllFoo() Admin APIs.

Reply:
Hmm, unfortunately I missed this in the other places. We should not be mixing `query` with `limit` and `offset` and `sort`; each of these are separate options. The `query` should have the raw database query in it -- which should be passed to `find` without a need for the extra parameter massaging `formattedQuery` helpers stuff.

We should not let this pattern get further adopted. We should do the right thing and separate out `limit` and `offset` and `sort`, etc. That way we don't have to do special helpers to reshuffle things to what they should be. We'll want to fix that anywhere else that adopted the bad pattern at some point.

So we should be doing: `getAll({query, limit, offset, explain = false} = {})`.

Reply:
@dlongley Sorry, I accidentally edited your comment when trying to reply. I thought I had `quote reply` selected 😓. I couldn't find a way to revert it so I just copy pasted the original comment. 

I've separated out `limit` and `offset` and added them as separate options in this commit [71a2a83](https://github.com/digitalbazaar/bedrock-resource-restriction/pull/47/commits/71a2a83b298ab87ba1b92d7ded9022abf6cf26f0). 

> We'll want to fix that anywhere else that adopted the bad pattern at some point.

Noted. Whenever I get a chance, I'll fix that incorrect pattern in other places.

Reply:
I have created an issue in the other repo where this is present.

Reply:
@mandyvenables thanks Mandy

Reply:
@JSAssassin no worries about the accidental edit! :)